thorlcr/graph/thgraph.hpp

```diff
@@ -1121,7 +1117,7 @@ class graph_decl CActivityBase : implements CInterfaceOf<IThorRowInterfaces>, im
     CSingletonLock CABserializerlock;
     CSingletonLock CABdeserializerlock;
     roxiemem::RoxieHeapFlags defaultRoxieMemHeapFlags = roxiemem::RHFnone;
-    Owned<CFileSizeTracker> tempFileSizeTracker;
+    mutable OwnedSingleton<CFileSizeTracker> tempFileSizeTracker;
```

Comment:
A slight concern about the extra increase in object size (48bytes or a lot more on macos).  Not a big concern for this class... but it looks like a very light-weight member.

A class where the critical section is passed into the query function would avoid that problem (similar to the querySingleton() function.

Another possibility would be to add a similar query() function to AtomicShared, with a critical section parameter.